src/util/prolog/testPrograms.ts

```diff
@@ -43,4 +43,23 @@ export const familySession = new PrologSession(`
   grandmother(X, Z):-
   grandparent(X, Z),
   female(X).
-    `, 1000);
+  `, 1000);
+
+export const max = new PrologSession(`
+  max(X,Y,Y)  :-  X  =<  Y, !.
+  max(X, Y, X).
+  `, 1000);
```

Comment:
How comes you're always setting the step limit to 1000? Do you strictly need it or could you omit it? If it's just some "large number that will surely never get hit and I only use it so my tests will always terminate", please create a constant for that ;)